src/fhirtypes/ElementDefinition.ts

```diff
@@ -1962,10 +1962,23 @@ export class ElementDefinition {
    *   be resolved.
    */
   unfold(fisher: Fishable): ElementDefinition[] {
-    if (
-      (this.type?.length === 1 && (this.type[0].profile ?? []).length <= 1) ||
-      this.contentReference
-    ) {
+    // if we have a choice element where id ends with [x], proceed if there is at most one profile, since it expects to get [] back in that case
+    // if we don't have a choice element, proceed with any number of profiles
+    const isChoiceElement = this.id.endsWith('[x]');
```

Comment:
I don't think I fully understand why we wouldn't want to take this same approach for choice elements _if_ they have already been constrained down to one type?  E.g., if this was done:
```
* value[x] only MyQuantityProfile1 or MyQuantityProfile2
```
then even though it is a choice (e.g., ends in `[x]`), it only has one type (`Quantity`), so at this point it acts like a single-type element w/ two profiles. And I think it would be appropriate to treat it as such -- issuing a warning and then using the base type (`Quantity`).

Reply:
Yes, I think that makes sense. So, a choice element with multiple types shouldn't be unfolded here, but a choice element with one type should proceed with unfolding (and issue a warning if there are multiple profiles on that type). So, when the choice element has multiple types and ends up in `unfoldChoiceElementTypes`, should the same warning be issued if any of those types have multiple profiles on them? The current behavior is to just always use the common ancestor type among all of the types and profiles (which might end up just being `Element`), and that will still happen, just maybe with a warning on top if one of the types has multiple profiles on it.

Reply:
Re: unfolding a choice w/ multiple types... it seems that if we already have to unfold at a higher level than a single type, that maybe the multiple profiles really doesn't matter as much anyway.  I mean, if the choice is A or B and B, I don't think our logic changes whether or not either one has multiple profiles, right?  Because the common ancestor is higher than the `type.code`s represented?  So in that case, maybe a warning is appropriate in general, but not necessarily because of the profiles.  But I feel like that's starting to go out of scope of this PR, so I wouldn't really worry about it...